### PR TITLE
improvement(forking): unlink, settings page migration, deployed chats, public apis, external mcp servers/tools, special subblocks

### PR DESCRIPTION
**Pull request by icecrasher321**
## Summary

Addresses UX issues by migrating to a settings page. Solves edge cases found during testing. 

## Type of Change
- [x] Bug fixes
- [x] Other: UX

## Testing
Tested manually 

## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my changes
- [x] Tests added/updated and passing
- [x] No new warnings introduced
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

### CODE REVIEW

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
bugbot run

File: apps/sim/ee/workspace-forking/components/fork-activity-panel/fork-activity-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      emptyState={
+        isError ? (
+          <SettingsEmptyState variant='inline'>
+            Couldn't load activity. Try again shortly.
+          </SettingsEmptyState>
+        ) : isPending ? undefined : (
+          <SettingsEmptyState variant='inline'>
+            Nothing here yet. Forks, syncs, and rollbacks will appear here.
+          </SettingsEmptyState>
+        )
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/ee/workspace-forking/components/fork-activity-panel/fork-activity-panel.tsx:301]: intentional

File: apps/sim/ee/workspace-forking/components/fork-sync/use-fork-sync.ts [hunk 2]
```diff
@@ -0,0 +1,810 @@
...
+          direction,
+          // Once the diff has loaded, ALWAYS send the full effective set - including `[]`,
+          // which means "every dependent went away" and must reconcile/clear the live replace
+          // targets' stored rows. Collapsing `[]` into omission would make the backend
+          // PRESERVE stale rows. Only omit before the diff loads (set unknown), so the
+          // existing store is left untouched.
+          ...(dependentValues !== null ? { dependentValues } : {}),
+          ...(selectedCopyables.length > 0 ? { copyResources } : {}),
+        },
+      })
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/ee/workspace-forking/components/fork-sync/use-fork-sync.ts:716]: promote resolves references through the persisted mapping store, so entries must be saved first; mappings are durable edge config (same as the Save button), while dependentValues/copyResources apply transactionally inside the promote itself."

**Comment by icecrasher321** [comment]
@greptile

**Comment by icecrasher321** [comment]
bugbot run
